solr/core/src/java/org/apache/solr/response/transform/ChildDocTransformerFactory.java

```diff
@@ -179,7 +179,11 @@ public static BitSetProducer getCachedBitSetProducer(final SolrQueryRequest requ
     SolrCache<Query, BitSetProducer> parentCache = request.getSearcher().getCache(CACHE_NAME);
     // lazily retrieve from solr cache
     if (parentCache != null) {
-      return parentCache.computeIfAbsent(query, QueryBitSetProducer::new);
+      try {
+        return parentCache.computeIfAbsent(query, QueryBitSetProducer::new);
+      } catch (IOException e) {
+        throw new AssertionError("Can't happen");
```

Comment:
Not a fan of this myself.ll Impls change, new impls are added, it's declared to throw an IOException. Personally, I'd just turn it into a runtime io exception.